Mathlib/RingTheory/Support.lean

```diff
@@ -0,0 +1,204 @@
+/-
+Copyright (c) 2024 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.RingTheory.PrimeSpectrum
+import Mathlib.Algebra.Module.LocalizedModule
+import Mathlib.RingTheory.Localization.AtPrime
+
+/-!
+
+# Support of a module
+
+## Main results
+- `Module.support`: The support of an `R`-module as a subset of `Spec R`.
+- `Module.mem_support_iff_exists_annihilator`: `p ∈ Supp M ↔ ∃ m, Ann(m) ≤ p`.
+- `Module.stableUnderSpecialization_support`: `Supp M` is stable under specialization.
+- `Module.support_eq_empty_iff`: `Supp M = ∅ ↔ M = 0`
+- `Module.support_of_exact`: `Supp N = Supp M ∪ Supp P` for an exact sequence `0 → M → N → P → 0`.
+- `Module.support_eq_zeroLocus`: If `M` is `R`-finite, then `Supp M = Z(Ann(M))`.
+- `LocalizedModule.exists_subsingleton_away`:
+  If `M` is `R`-finite and `Mₚ = 0`, then `M[1/f] = 0` for some `p ∈ D(f)`.
+
+Also see `AlgebraicGeometry/PrimeSpectrum/Module` for other results
+depending on the zariski topology.
+
+## TODO
+- Connect to associated primes once we have them in mathlib.
+- Given an `R`-algebra `f : R → A` and a finite `R`-module `M`,
+  `Supp_A (A ⊗ M) = f♯ ⁻¹ Supp M` where `f♯ : Spec A → Spec R`. (stacks#0BUR)
+-/
+
+-- Basic files in `RingTheory` should avoid depending on the Zariski topology
+-- See `AlgebraicGeometry/PrimeSpectrum/Module`
+assert_not_exists TopologicalSpace
+
+variable {R M} [CommRing R] [AddCommGroup M] [Module R M]
```

Comment:
```suggestion
variable {R M : Type*} [CommRing R] [AddCommGroup M] [Module R M]
```

Reply:
Is there a reason to repeat `{p}` so much instead of `{p : PrimeSpectrum R}` here?

Reply:
Fixed. Thanks!